cmd/internal/pluginutils/bypath.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginutils
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+const (
+	pciSlotPrefix = "PCI_SLOT_NAME="
+)
+
+// Returns a slice of by-path Mounts for a pci device that is identified by the
+// uevent file contents. by-path files are searched from the given bypathDir.
+// In the by-path dir, any files that start with "pci-<pci bus>" will be added to mounts.
```

Comment:
hmm I wonder if reading `uevent` is the best approach for finding the PCI addr. Is that not available through the sysfs links?

Reply:
Not that I could find. With some other devices I've seen symlinks to the pci bus but that doesn't appear to be the case with the i915 device. @eero-t do you know?

Reply:
No idea unfortunately.

Reply:
ok, it's a bit more complex on the server I checked:
$ readlink /sys/class/drm/card0
../../devices/pci0000:4a/0000:4a:02.0/0000:4b:00.0/0000:4c:01.0/0000:4d:00.0/drm/card0

But seems like parsing the last address part would work.

Reply:
Implementation is now using the symlink.